### PR TITLE
Fix Namespace for MachineHealthCheck Objects

### PR DESCRIPTION
**Pull request by mdsgabriel**
Modify the template that generates MachineHealthChecks (MHC) to use the
eksa-system namespace (the same used by other CRs).

During cluster creation if the MHCs are created on a separate namespace they will not be available.

*Issue #, if available:*

*Description of changes:*

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by abhay-krishna** [comment]
[APPROVALNOTIFIER] This PR is **APPROVED**

This pull-request has been approved by: *<a href="https://github.com/aws/eks-anywhere/pull/303#" title="Author self-approved">mdsgabriel</a>*

The full list of commands accepted by this bot can be found [here](https://go.k8s.io/bot-commands?repo=aws%2Feks-anywhere).

The pull request process is described [here](https://git.k8s.io/community/contributors/guide/owners.md#the-code-review-process)

<details >
Needs approval from an approver in each of these files:

- ~~[OWNERS](https://github.com/aws/eks-anywhere/blob/main/OWNERS)~~ [mdsgabriel]

Approvers can indicate their approval by writing `/approve` in a comment
Approvers can cancel approval by writing `/approve cancel` in a comment
</details>

File: pkg/providers/vsphere/vsphere_test.go
Comment (abhay-krishna) [inline pkg/providers/vsphere/vsphere_test.go:2189]: nit: this can be refactored to
```YAML
namespace: %[1]v
...
namespace: %[1]v
...
`, constants.EksaSystemNamespace)
```

Comment (mdsgabriel) [reply pkg/providers/vsphere/vsphere_test.go]: done

**Review by abhay-krishna — APPROVED** [review]
/lgtm

**Pull request merged**
